src/lib/Export.hs

```diff
@@ -143,6 +143,10 @@ prepareFunctionForExport env nameStr func = do
           elemTy' <- substBuilder (b@>Var i) elemTy
           createDest (idx <> Nest (Bind i) Empty) elemTy'
         return (Con $ TabRef destTab, exportResult)
+      PairTy a b -> do
```

Comment:
Can you add a check that makes sure that pair types only appear on the outermost level of recursion? I think it's enough to add `| idx == Empty` after `PairTy a b` (to add an extra guard that we're not recursing through any table arrows).

Reply:
Yep, done.